src/client/providers/jediProxy.ts

```diff
@@ -334,250 +381,201 @@ function spawnProcess(dir: string) {
+                            const cmd = this.commands.get(id);
+                            try {
+                                cmd.deferred.resolve(null);
+                            }
+                            catch (ex) {
+                            }
+                            this.commands.delete(id);
                         }
-                        commands.delete(id);
-                    }
-                });
-            }
+                    });
+                }
+            });
         });
-    });
-}
-
-function sendCommand<T extends ICommandResult>(cmd: ICommand<T>): Promise<T> {
-    if (!proc) {
-        return Promise.reject(new Error("Python proc not initialized"));
     }
-    var executionCmd = <IExecutionCommand<T>>cmd;
-    var payload = createPayload(executionCmd);
-    executionCmd.deferred = createDeferred<T>();
-    // if (typeof executionCmd.telemetryEvent === 'string') {
-    //     executionCmd.delays = new telemetryHelper.Delays();
-    // }
-    try {
-        proc.stdin.write(JSON.stringify(payload) + "\n");
-        commands.set(executionCmd.id, executionCmd);
-        commandQueue.push(executionCmd.id);
-    }
-    catch (ex) {
-        console.error(ex);
-        //If 'This socket is closed.' that means process didn't start at all (at least not properly)
-        if (ex.message === "This socket is closed.") {
 
-            killProcess();
+    public sendCommand<T extends ICommandResult>(cmd: ICommand<T>): Promise<T> {
+        if (!this.proc) {
+            return Promise.reject(new Error("Python proc not initialized"));
         }
-        else {
-            handleError("sendCommand", ex.message);
+        var executionCmd = <IExecutionCommand<T>>cmd;
+        var payload = this.createPayload(executionCmd);
+        executionCmd.deferred = createDeferred<T>();
+        // if (typeof executionCmd.telemetryEvent === 'string') {
+        //     executionCmd.delays = new telemetryHelper.Delays();
+        // }
+        try {
+            this.proc.stdin.write(JSON.stringify(payload) + "\n");
+            this.commands.set(executionCmd.id, executionCmd);
+            this.commandQueue.push(executionCmd.id);
         }
-        return Promise.reject(ex);
-    }
-    return executionCmd.deferred.promise;
-}
-
-function createPayload<T extends ICommandResult>(cmd: IExecutionCommand<T>): any {
-    var payload = {
-        id: cmd.id,
-        prefix: "",
-        lookup: commandNames.get(cmd.command),
-        path: cmd.fileName,
-        source: cmd.source,
-        line: cmd.lineIndex,
-        column: cmd.columnIndex,
-        config: getConfig()
-    };
-
-    if (cmd.command === CommandType.Symbols) {
-        delete payload.column;
-        delete payload.line;
-    }
-
-    return payload;
-}
+        catch (ex) {
+            console.error(ex);
+            //If 'This socket is closed.' that means process didn't start at all (at least not properly)
```

Comment:
Not going to block approval, but missing a period at the end (and on most other comments).

Reply:
Thanks, for once we get over the multiroot, please do block approval. I want code when adding multi root support.